src/invoice.ts

```diff
@@ -8,10 +8,29 @@ export default class Invoice {
   paymentHash: string;
   preimage: string | null;
   verify: string | null;
+  satoshi: number;
+  expiry: number; // expiry in seconds (not a timestamp)
+  timestamp: number; // created date in seconds
+  createdDate: Date;
+  expiryDate: Date;
+  description: string | null;
 
   constructor(args: InvoiceArgs) {
     this.paymentRequest = args.pr;
-    this.paymentHash = getHashFromInvoice(this.paymentRequest) as string;
+    if (!this.paymentRequest) {
+      throw new Error("Invalid payment request");
+    }
+    const decodedInvoice = decodeInvoice(this.paymentRequest);
+    if (!decodedInvoice) {
+      throw new Error("Failed to decode payment request");
+    }
+    this.paymentHash = decodedInvoice.paymentHash;
+    this.satoshi = decodedInvoice.satoshi;
+    this.timestamp = decodedInvoice.timestamp;
```

Comment:
should we expose timestamp and expiry? or just the date objects? (I think these are easier to use in JS)